### PR TITLE
ws_eink: update cleanup resource for driver

### PR DESCRIPTION
**Pull request by nxthongbk**
Remove free cmap because it is not initialized before and add free defferred io

### CODE REVIEW

**Comment by nxthongbk** [comment]
Hi David 

I just update cleanup resource.
Because we using vzalloc so in cleanup we need to use vfree instead of kfree.

Beside I added cleanup deffered io to make sure all resouce is cleaned. I tested and driver worked properly

Thanks
Thong Nguyen

File: experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c [hunk 1]
```diff
@@ -606,27 +606,22 @@ static int ws_eink_spi_probe(struct spi_device *spi)
...
 	unregister_framebuffer(p);
-	/*
-	 * TODO: Why is this necessary? It doesn't seem that there is a call to
-	 * fb_alloc_cmap()
-	 */
-	fb_dealloc_cmap(&p->cmap);
 	iounmap(p->screen_base);
-	kfree(p->screen_base);
-	kfree(par->ssbuf);
+	vfree(par->ssbuf);
```
Comment (dpfrey) [inline experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c]: You changed `kfree(par->ssbuf)` into `vfree(par->ssbuf)`.  I guess that makes sense because it was allocated using `vzalloc()`.  Why did you remove `kfree(p->screen_base)` rather than change it to `vfree(p->screen_base)`?

**Comment by nxthongbk** [comment]
hi David

We used iounmap(p->screen_base) so we dont need to use vfree.

thanks
Thong Nguyen


Vào 11:27 AM, T.5, 21 Th3, 2019 David Frey <notifications@github.com> đã
viết:
